aggoracle/chaingersender/evm.go

```diff
@@ -99,32 +166,89 @@ func (c *EVMChainGERSender) IsGERInjected(ger common.Hash) (bool, error) {
 	return gerIndex.Cmp(common.Big0) == 1, nil
 }
 
+// IsGERProposed checks if the provided global exit root has already been proposed by the oracle committee member
+func (c *EVMChainGERSender) IsGERProposed(ger common.Hash) (bool, error) {
+	if c.mode != AggOracleCommitteeMode {
+		return false, fmt.Errorf("IsGERProposed is only available in AggOracleCommittee mode, current mode: %s", c.mode)
+	}
+
+	lastProposedGER, err := c.aggOracleCommittee.AddressToLastProposedGER(
+		&bind.CallOpts{Pending: false}, c.ethTxMan.From())
+	if err != nil {
+		return false, fmt.Errorf("failed to check last proposed GER for oracle committee member %s: %w",
+			c.ethTxMan.From(), err)
+	}
+
+	lastProposedGERHash := common.Hash(lastProposedGER)
+	return lastProposedGERHash == ger, nil
```

Comment:
What happens if you propose a GER-1, then propose GER-2, then try to propose again GER-1? Its not something that will happen with honest actors, but what happens if someone tries to do this as a malicious actor?